### PR TITLE
allow custom exchange type + add strategy design features to all trading modes

### PR DESCRIPTION
**Pull request by techfreaque**

### CODE REVIEW

File: Services/Interfaces/web_interface/flask_util/file_services.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 def send_and_remove_file(file_path, attachment_filename):
     try:
-        return flask.send_file(file_path, as_attachment=True, attachment_filename=attachment_filename, cache_timeout=0)
+        return flask.send_file(file_path, as_attachment=True, download_name=attachment_filename, max_age=0)
```
Comment (GuillaumeDSM) [inline Services/Interfaces/web_interface/flask_util/file_services.py:23]: thanks for the fix, however could you rebase of dev please ? This is fixed in the dev branch :)

File: Trading/Mode/scripted_trading_mode/scripted_trading.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import octobot_trading.modes as modes
+
+
+class ScriptedTradingMode(modes.AbstractScriptedTradingMode):
```
Comment (Herklos) [inline Trading/Mode/scripted_trading_mode/scripted_trading.py:19]: I don't think we should add it yet

File: Trading/Mode/arbitrage_trading_mode/arbitrage_trading.py [hunk 3]
```diff
@@ -47,13 +47,15 @@ def get_current_state(self) -> (str, float):
                self.producers[0].final_eval if self.producers[0].final_eval else "N/A"
 
     async def create_producers(self) -> list:
+        default_producers = await super().create_producers()
```
Comment (GuillaumeDSM) [inline Trading/Mode/arbitrage_trading_mode/arbitrage_trading.py:50]: Why do we need to call `await super().create_producers()` in every trading mode now ?

**Review by Herklos — APPROVED** [review]
Nice!

File: Trading/Mode/arbitrage_trading_mode/arbitrage_trading.py (hunk 3 above)
Comment (techfreaque) [reply Trading/Mode/arbitrage_trading_mode/arbitrage_trading.py]: I moved it to the trading_modes_updates branch and changed it to self.create_default_consumers() and self.create_default_producers()

**Pull request merged**